From 21c6faa9fa294c0fd5199d71fba08abc9ee32f96 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 26 Jun 2016 15:20:22 +0300
Subject: [PATCH 14/24] ARM: i.MX6: dts: enable mipi_dsi

Enable mipi_dsi.
Add a separate fx6-sbc dts file with a mipi port enabled.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm/boot/dts/imx6q-sbc-fx6-mipi.dts | 44 ++++++++++++++++++++++++++++++++
 arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi    |  9 -------
 2 files changed, 44 insertions(+), 9 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6q-sbc-fx6-mipi.dts

diff --git a/arch/arm/boot/dts/imx6q-sbc-fx6-mipi.dts b/arch/arm/boot/dts/imx6q-sbc-fx6-mipi.dts
new file mode 100644
index 0000000..c591819
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-sbc-fx6-mipi.dts
@@ -0,0 +1,44 @@
+/*
+* Copyright 2014 CompuLab Ltd.
+*
+* Author: Valentin Raevsky <valentin@compulab.co.il>
+*
+* The code contained herein is licensed under the GNU General Public
+* License. You may obtain a copy of the GNU General Public License
+* Version 2 or later at the following locations:
+*
+* http://www.opensource.org/licenses/gpl-license.html
+* http://www.gnu.org/copyleft/gpl.html
+*/
+
+/dts-v1/;
+#include "imx6q.dtsi"
+#include "imx6q-cm-fx6.dtsi"
+#include "imx6qdl-sb-fx6x.dtsi"
+#include "imx6qdl-sb-fx6.dtsi"
+
+/ {
+	model = "CompuLab CM-FX6 on SBC-FX6";
+	compatible = "compulab,cm-fx6", "compulab,sbc-fx6", "fsl,imx6q";
+};
+
+&ipu1di0_disp {
+	status = "disabled";
+};
+
+&mipi_dsi {
+	dev_id = <0>;
+	disp_id = <0>;
+	lcd_panel = "TRULY-WVGA";
+	depends = &pca9555;
+	disp-power-on-supply = <&reg_mipi_dsi_pwr_on_mode>;
+	resets = <&mipi_dsi_reset>;
+	status = "okay";
+};
+
+&mxcfb1 {
+	mode_str ="TRULY-WVGA";
+	disp_dev = "mipi_dsi";
+	depends = &mipi_dsi;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
index 97fc1af..e96b732 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
@@ -183,12 +183,3 @@
 	status = "okay";
 };
 
-&mipi_dsi {
-	dev_id = <0>;
-	disp_id = <0>;
-	lcd_panel = "TRULY-WVGA";
-	depends = "nxp,pca9555";
-	disp-power-on-supply = <&reg_mipi_dsi_pwr_on_mode>;
-	resets = <&mipi_dsi_reset>;
-	status = "disabled";
-};
-- 
1.9.1

